### PR TITLE
Add nuget metadata inside csproj files to pack using msbuild targets

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: adal/build/Microsoft.IdentityModel.Clients.ActiveDirectory.nuspec [hunk 1]
```diff
@@ -1,67 +0,0 @@
-﻿<?xml version="1.0"?>
```
Comment (bgavrilMS) [inline adal/build/Microsoft.IdentityModel.Clients.ActiveDirectory.nuspec:1]: Nuspec is no longer needed to package

File: adal/src/Directory.Build.targets [hunk 2]
```diff
@@ -1,46 +0,0 @@
-<Project>
```
Comment (bgavrilMS) [inline adal/src/Directory.Build.targets:1]: Not seem to be used / imported anywhere - https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/search?q=Directory.Build.targets&unscoped_q=Directory.Build.targets

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Microsoft.IdentityModel.Clients.ActiveDirectory.csproj [hunk 3]
```diff
@@ -1,13 +1,5 @@
 ﻿<Project Sdk="MSBuild.Sdk.Extras/1.5.4">
 
-  <!-- Exclude the analysis of this framework type: workaround for https://jira.sonarsource.com/browse/SONARMSBRU-317 -->
```
Comment (bgavrilMS) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Microsoft.IdentityModel.Clients.ActiveDirectory.csproj:3]: Dead code until we decide to use SQ again.

**Review by jennyf19 — APPROVED** [review]
Nice clean up. :shipit:

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Microsoft.IdentityModel.Clients.ActiveDirectory.csproj (hunk 3 above)
Comment (jmprieur) [reply adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Microsoft.IdentityModel.Clients.ActiveDirectory.csproj:3]: and I think that SQ fixed it also

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Microsoft.IdentityModel.Clients.ActiveDirectory.csproj [hunk 4]
```diff
@@ -20,17 +12,28 @@
...
     <Company>Microsoft Corporation</Company>
     <Product>Active Directory Authentication Library</Product>
+    <Authors>Microsoft</Authors>
+    <PackageLicenseUrl>https://go.microsoft.com/fwlink/?linkid=841311</PackageLicenseUrl>
+    <PackageProjectUrl>https://aka.ms/adalnet</PackageProjectUrl>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+    <Description>
+      This package contains the binaries of the Active Directory Authentication Library (ADAL). ADAL provides easy to use authentication functionality for your .NET based client by taking advantage of Windows Server Active Directory and Azure Active Directory.
+      ADAL.NET enables you to acquire a security token to access protected Web APIs, for instance Microsoft Graph or your own Web API.
+      ADAL.NET is a .NET Standard class library which you can consume on various .NET client platforms (including Windows desktop, UWP, Windows 8.1, Xamarin iOS and Xamarin Android) but also you Web applications and Web APIs (ASP.NET, .NET Core, ASP.NET Core) that call other Web APIs in the name of a user, or without a user.
```
Comment (jmprieur) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Microsoft.IdentityModel.Clients.ActiveDirectory.csproj]: it's more than a .NET Standard class library now,

Comment (jmprieur) [reply adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Microsoft.IdentityModel.Clients.ActiveDirectory.csproj]: the platform have changed:

(including Windows desktop (.NET 4.5+ and .NET Core), ~~Windows 8.1,~~ UWP/Windows 10, Xamarin iOS and Xamarin Android) but also you Web applications and Web APIs (ASP.NET, ASP.NET Core)

File: core/src/Microsoft.Identity.Core.csproj [hunk 5]
```diff
@@ -19,17 +11,9 @@
...
-    <!-- This will generate AssemblyVersion, AssemblyFileVersion and AssemblyInformationVersion -->
-    <Version>$(CoreSemVer)</Version>
-
-    <Copyright>Copyright (c) Microsoft Corporation. All rights reserved.</Copyright>
-    <Company>Microsoft Corporation</Company>
-  </PropertyGroup>
-
+<!-- Important note: this assembly should not be shipped to customers; it acts only as a placeholder for files that are included
+in ADAL and MSAL and to ease unit testing. No projects should take a dependency on Core.dll either -->
+
```
Comment (jmprieur) [inline core/src/Microsoft.Identity.Core.csproj:16]: Is there a way to ensure that no project take a dependency on core.dll?

**Review by jmprieur — CHANGES_REQUESTED** [review]
LGTM
I've requested a small change (remove windows 8)

File: core/src/Microsoft.Identity.Core.csproj (hunk 5 above)
Comment (bgavrilMS) [reply core/src/Microsoft.Identity.Core.csproj:16]: Layer2 analyzer? We could move all core unit tests to either ADAL or MSAL (or both) and make core.dll either a shared project or a .net standard 4.7 project, on which ADAL and MSAL cannot take a reference.

**Review by jmprieur — APPROVED** [review]
LGTM

**Pull request merged**